src/rpc/server/connection.zig

```diff
@@ -0,0 +1,264 @@
+const builtin = @import("builtin");
+const std = @import("std");
+
+pub const GetSockNameError = std.posix.GetSockNameError;
+pub fn getSockName(
+    socket_handle: std.posix.socket_t,
+) GetSockNameError!std.net.Address {
+    var addr: std.net.Address = .{ .any = undefined };
+    var addr_len: std.posix.socklen_t = @sizeOf(@TypeOf(addr.any));
+    try std.posix.getsockname(socket_handle, &addr.any, &addr_len);
+    return addr;
+}
+
+/// When this is false, it means `accept[Handled]` can't apply
+/// flags to the accepted socket, and the caller will have to
+/// to ensure relevant flags are enabled/disabled after acceptance.
+pub const have_accept4 = !builtin.target.isDarwin();
```

Comment:
screaming case ? and at top of file

Reply:
706f136b75d03abed6a66023acce4c00a76e85c2